### ISSUE TITLE
[typescript-eslint/no-unsafe-call] false positive in vs code when editing

### ISSUE THREAD

**Issue by sfabriece** [typescript-eslint/typescript-eslint#2885]
- [ ] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-call": ["error"]
  }
}
```

```TS
if (!(await this.repository.exists(id))) {
      throw new NotFound();
 }
```

tsconfig
```
{
  "compilerOptions": {
    "module": "CommonJS",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    // strict checks
    "alwaysStrict": true,
    "noImplicitAny": true,
    "noImplicitThis": true,
    "strict": true,
    "strictPropertyInitialization": false,
    // Linter Checks
    "noFallthroughCasesInSwitch": true,
    "noImplicitReturns": true,
    "noUnusedLocals": false,
    "noUnusedParameters": true,
    // advanced
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,

    "target": "ES2020",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "lib": ["ES2020"]
  },
  "exclude": [""]
}

``` 

**Expected Result**

I expected eslint to not report an error indicating that `.exist()` is typed to any even though it is typed to `Promise<boolean>`

**Actual Result**

A lint error in vs code. Actually running eslint cli does not produce the error

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.10.0` |
| `@typescript-eslint/parser`        | `4.10.1` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.15.0` |
| `node`                             | `15.0.1` |
| 'eslint vs code plugin'      |`2.1.14`|

**Comment by bradzacher** [comment]
Have you tried restarting your IDE?
You didn't tick the box.

Sometimes things can get out of sync which means lint rules might receive the wrong types.

**Comment by sfabriece** [comment]
@bradzacher this error only happens while editing as stated in the title. As soon as the IDE is restarted the error does not show up again.

**Comment by bradzacher** [comment]
Take a look at your issue.
How much information have you given?
You have a tsconfig and a code ***fragment*** - a fragment without any additional types or information.

I can't help you yet.
You're a developer - you know what's required to debug issues.

# Help me to help you.

**Comment by sfabriece** [comment]
My apologies. I'll provide a proper reproduction.

**Comment by sfabriece** [comment]
I have been trying to set a test project but I'm having trouble reproducing the errors. It might have something to do with the size of the project. 

Unfortunately I can't share the code of the project I'm working on publicly. Any suggestions on how I can help you help me here @bradleyayers

**Comment by bradzacher** [comment]
Without a repro or any more information about what your code/type looks like - it's hard for me to provide any help.

**Comment by krisztianb** [comment]
I'm also experiencing this issue from time to time and can confirm the behaviour @sfabriece has (vaguely) described.

**Comment by sfabriece** [comment]
I'll try and do a screencast, but I don't know how much help that is.

@krisztianb can you help to describe the problem in your own words?

**Comment by krisztianb** [comment]
@sfabriece I'll be offline for a couple of week but after that might play around and try to come up with a way to reproduce this issue.

**Comment by bradzacher** [comment]
I'm going to close this issue as it's inactionable.
If you manage to create a repro - feel free to comment back, or raise a new issue.
